pkg/stream/consumer_test.go

```diff
@@ -282,6 +282,36 @@ var _ = Describe("Streaming Consumers", func() {
 
 	})
 
+	It("commit at flush interval with constant stream of incoming messages", func() {
+		producer, err := env.NewProducer(streamName, nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		var messagesReceived int32 = 0
+		consumer, err := env.NewConsumer(streamName,
+			func(consumerContext ConsumerContext, message *amqp.Message) {
+				atomic.AddInt32(&messagesReceived, 1)
+			}, NewConsumerOptions().
+				SetAutoCommit(NewAutoCommitStrategy().
+					SetCountBeforeStorage(10000000).
+					SetFlushInterval(time.Second)))
+		Expect(err).NotTo(HaveOccurred())
+
+		maxMessages := 50
+		for i := 0; i < maxMessages; i++ {
+			Expect(producer.Send(CreateMessageForTesting("", i))).NotTo(HaveOccurred())
+			// emit message before the flush interval has elapsed
+			time.Sleep(time.Millisecond * 100)
+
+			if consumer.GetLastStoredOffset() > 0 {
+				break
+			}
+		}
+
+		Expect(messagesReceived > 5 && messagesReceived < int32(maxMessages)).To(BeTrueBecause("%d messages received", messagesReceived))
+		Expect(producer.Close()).NotTo(HaveOccurred())
+		Expect(consumer.Close()).NotTo(HaveOccurred())
```

Comment:
```suggestion
```

Reply:
Consumer has to be closed no? The line above is closing the producer, so close() is not called twice on the consumer :)

Reply:
Oh, I thought it was a duplicate line, sorry :sweat_smile: